src/couch/test/couch_doc_json_tests.erl

```diff
@@ -269,6 +269,12 @@ from_json_error_cases() ->
             {doc_validation, "RevId isn't a string"},
             "Revision ids must be strings."
         },
+        {
+            {[{<<"_revisions">>, {[{<<"start">>, 0},
+                {<<"ids">>, [<<"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx">>]}]}}]},
+            {doc_validation, "RevId isn't a valid hexadecimal"},
+            "Revision ids must be a valid hex."
```

Comment:
Since when is this an invariant?

Reply:
And it's not about being valid hex, but being a valid MD5 hex hash

Reply:
Sorry, I'm not sure what are you trying to ask. A doc's revision been [md5 hash prefixed with doc's update count](https://docs.couchdb.org/en/latest/intro/api.html#revisions) for a pretty long time and as you can see this is documented.

This test _is_ kind of about validity of hex. Technically when `new_edits` set to `false` revision can be anything, but if it happens to be 16 bytes long rev's parser trying to treat it as a hex number, hence [the original issue](https://github.com/apache/couchdb/issues/1895).

Again, sorry, it might be just my perception, but your comments seems to be somehow aggressive. Is there a particular reason you are leaving them on this half-year old PR? If you have a problem with it, can you, please, open a new issue and describe it?

Reply:
Um, I'm not sure how that's aggressive. Anyway it's not meant to be in anyway.

I was going through the changelog for 2.3 and ended up on this PR.

>  but if it happens to be 16 bytes long rev's parser trying to treat it as a hex number

Yes, this has been a huge problem for us (https://github.com/apache/couchdb/issues/2015) and I thought this PR made it worse. This strange behaviour still only applies if it's 16 bytes long? 

Anyway I still think this quirk should be documented since even though custom rev ids are discouraged they are allowed.

Reply:
Look at this from my perspective: a stranger came out of nowhere and with zero context complaint about a message on a unit test of my ancient PR. This is a bit unusual for me and didn't seem to be well-intended. Glad to hear this is not the case.

Anyway, this PR does what it says in description, it trades a crash with `badarg` message for an error that, I hoped, would hint on a reason for such crash. It doesn't change rev's parsing logic in any way.

> This strange behaviour still only applies if it's 16 bytes long? i.e. not all revids must be a hex.

In short - yes. It distills from the fact that `new_edits=false` is a technical feature (in sense akin Apple's private APIs) that turns off system's assumption about MVCC, but if sent revision _looks_ like a "real" one it tricks system into applying some of these assumptions back. I see it was discussed pretty extensively on the issue you linked, not sure if I can add anything new here.

> Anyway I still think this quirk should be documented since even though custom rev ids are discouraged they are allowed.

Maybe? Sorry, it feels strange to discuss it in a middle of a merged PR. I suggest you to nudge on the original issue or even just change the documentation as you find desired and open a PR for it, it's an open source project after all.